### DISCUSSION TITLE
spell checking doesn't work when I switch from linux to windows

### DISCUSSION THREAD

**Discussion by Karduin** [vkbo/novelWriter#1066] [Help] [answered]
Hi, first of all, thank you for this software.
I installed NovelWriter on my linux and set the spell checking to french. Everything works fine. I also installed NovelWriter on my Windows.
I git push / git pull the project created on linux to windows.
The settings are the same, I added the french dictionary as indicated. and add pyenchant to my python installation just in case.
But all words are underlined.

in fact even with a new project it doesn't work :-(

I don't know what's wrong, any idea?

EDIT:
How to know which version of python is launched by novelwriter ? I saw the 3.10 in the app folder.
I have a 3.9 that runs by default is it OK ?
enchant work on python 64bits ?

**Comment by vkbo** [comment, marked as answer]
If you install via the setup.exe installer, novelWriter is bundled with the dependencies and the embeddable Python package. I always release with the latest patch release of Python. If you launch novelWriter from the generated desktop or start menu icon, it should use the embedded python.exe.

Anyway, some questions. If you open Preferences, can you select French as the main language? If so, Enchant can find your dictionary. It should also say "French" on the status bar at the bottom of the main window.

It is of course possible that the dictionary is broken. If any words are underlined, the spell checker is definitely running but for some reason all words are rejected. Normally that indicates that the wrong language is selected.

Just to rule out that there is no bug that causes the wrong dictionary to be loaded. If you type something in English, is that too underlined?

**Reply by Karduin** [reply]
Yes i can select french and say french on the status bar.

Spell checker works fine with English and dictionnary english.

With french dictionnary english words are underlined

it seems that it is the dictionary that is at fault.

I took the files from https://cgit.freedesktop.org/libreoffice/dictionaries/tree/ fr_FR folder and then put in users/me/appdata/local/enchant/hunspell

I also check in a terminal and dictionnary fail.

```
>>> import enchant
>>> enchant.list_languages()
['en_HK', 'en_BW', 'en_BZ', 'en_GB', 'en_JM', 'en_DK', 'en_GH', 'en_US', 'fr', 'en_ZW', 'en_SG', 'en_ZA', 'en_NZ', 'en_IE', 'en_BS', 'en_AG', 'en_AU', 'en_NA', 'en_TT', 'en_IN', 'en_PH', 'en_NG', 'en_CA']
>>> d = enchant.Dict('en_US')
>>> d.check('hello')
True
>>> df = enchant.Dict('fr')
>>> df.check('bonjour')
False
>>> df.check('hello')
False
>>>
```

**Reply by vkbo** [reply]
Yeah, I just tried on Windows as well. The German dictionary works, but not the French. I have no idea why. I copied over the `fr.aff` and `fr.dic` files from `/usr/share/hunspell` on Linux, and it worked just fine.

It's clearly some issue with the dictionaries on free desktop.

**Reply by Karduin** [reply]
OK so it's windows plus fr that doesn't work ;-(

**Reply by vkbo** [reply]
Could be the format has changed?

I don't think it's a Windows issue. What I meant is that I copied the files from Linux to Windows, and it worked. You could also do that.

I want to add a tool in novelWriter to download the dictionary files and put them in the correct place for Windows (and perhaps Mac) users. But I need a reliable source to download them from. I was hoping freedesktop would be it, but clearly it isn't. Maybe I need to host a working copy of them all from a Linux source.

**Reply by Karduin** [reply]
thank you very much, I will do that

**Reply by vkbo** [reply]
Aha, downloading them from here worked as well: https://github.com/LibreOffice/dictionaries

Edit: I just forked the repository. I should be able to build a download tool from that. It should make these things easier for everyone. :)

**Reply by Karduin** [reply]
thank you very much, It's working now.

**Comment by Karduin** [comment]
Just one more thing, I got tricked because I right clicked download on fr.aff and fr.dict to download the file.

It's a big mistake because I get the html page. That's why it didn't work.

So you have to click on plain at the end of the corresponding line on the right and do file / save as.

So the link given to freedesktop in the doc works

sorry for the inconvenience

**Reply by vkbo** [reply]
I clicked on them and then saved the data, and still had issues, so that's strange. In any case, thanks for the update.
